### PR TITLE
Pass metadata from notebook to console

### PR DESCRIPTION
**Pull request by BoPeng**
What it does:  Allow `runInConsole` to pass cell metadata to the `promptCell` of console panel.

Why: In [SoS notebook](https://vatlab.github.io/sos-docs/), each cell in the notebook can have its own kernel, saved in `metadata['kernel']`. When `runInConsole` is called to execute the cell in the console panel, the console panel needs to know the kernel information in addition to the code to execute.

What has been done: This simple patch [sends `metadata`](https://github.com/vatlab/jupyterlab/blob/inject_metadata/packages/notebook-extension/src/index.ts#L983) as an optional parameter to [`CodeConsole.inject`](https://github.com/vatlab/jupyterlab/blob/inject_metadata/packages/console/src/widget.ts#L352).

NOTE: Tests pass locally but not on Azure etc. Note sure if they are related.

### CODE REVIEW

File: packages/console/src/widget.ts [hunk 1]
```diff
@@ -337,9 +337,14 @@ export class CodeConsole extends Widget {
    *
    * @returns A promise that indicates when the injected cell's execution ends.
    */
-  inject(code: string): Promise<void> {
+  inject(code: string, metadata: JSONObject = {}): Promise<void> {
     let cell = this.createCodeCell();
     cell.model.value.text = code;
+    if (metadata) {
```
Comment (jasongrout) [inline packages/console/src/widget.ts]: This is always true if the default is `{}`. How about we just delete the `if` check and let the `for` loop just iterate over an empty object if the default is passed in?

File: packages/console/src/widget.ts [hunk 2]
```diff
@@ -337,9 +337,14 @@ export class CodeConsole extends Widget {
...
+      for (let key in metadata) {
```
Comment (jasongrout) [inline packages/console/src/widget.ts]: FYI, in general iterating over an object using `in` can iterate over inherited properties you typically don't want to iterate over. More common patterns these days are to use `Object.keys(metadata).forEach(() => {cell.model.metadata.set(key, metadata[key]);});`, or even:

```typescript
for (let key of Object.keys(metadata)) {
         cell.model.metadata.set(key, metadata[key]); 
}
```

**Comment by jasongrout** [comment]
Great, thanks!

It looks like that last change introduced some changes from the previous version of prettier we were using. You merged master into this branch. You just need to update prettier by removing node_modules and doing another `jlpm install`, then those formatting reversions should go away.

**Comment by BoPeng** [comment]
Done. I had to include the linter suggested changes to push, did not realize the change of linters.

**Comment by jasongrout** [comment]
Thanks. The code looks good to me. I'm waiting for tests to pass.

**Pull request merged**